### PR TITLE
Rubix cube

### PR DESCRIPTION
**Pull request by R4PaSs**
A simple Rubix Cube library, don't mind the diff, a huge part is just test reuslts :)

It should support all the commands from the Singmaster notation (see [here](http://ruwix.com/the-rubiks-cube/notation/) for more details on the notation)

Note that the big number of specialized top-level functions with ints show in this case how useful a proper Enum system would be :)

### CODE REVIEW

File: lib/core/collection/array.nit [hunk 1]
```diff
@@ -301,6 +301,30 @@ abstract class AbstractArray[E]
...
+	end
+
+	# Rotates the elements of self once to the right
+	#
+	# ~~~nit
+	# var a = [12, 23, 34, 45]
+	# a.rrotate
+	# assert a == [45, 12, 23, 34]
+	# ~~~
+	fun rrotate do
```
Comment (privat) [inline lib/core/collection/array.nit]: can-you promote rotations to Sequence?

also, List and CircularArray shoud be a more efficient datastructure for roration. document this fact.

File: lib/rubix.nit [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+private fun bottom_ln: Int do return 2
+
+private fun left_col: Int do return 0
+private fun mid_col: Int do return 1
+private fun right_col: Int do return 2
+
+private fun clock: Int do return 0
+private fun counterclock: Int do return 1
+
+private fun square: String do return once 0x25a0.code_point.to_s
```
Comment (privat) [inline lib/rubix.nit]: why not `return "■"`?

File: lib/rubix.nit [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+		if self == 5 then return square.blue
+		abort
+	end
+end
+
+# In-memory representation of a Rubix Cube
+#
+# Faces are represented in memory as if they were a flattened cube like:
+#
+# ~~~verbatim
```
Comment (privat) [inline lib/rubix.nit]: `raw` not `verbatim`

File: lib/rubix.nit [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+		if not o isa RubixCube then return false
+		for mf in faces, tf in o.faces do
+			for ml in mf, tl in tf do
+				for mc in ml, tc in tl do if mc != tc then return false
+			end
+		end
+		return true
+	end
+
+	# Is `self` a solved Rubix Cube ?
```
Comment (privat) [inline lib/rubix.nit]: Are symmetries handled?

**Comment by privat** [comment]
nice +1

File: lib/core/collection/array.nit (hunk 1 above)
Comment (Morriar) [inline lib/core/collection/array.nit]: In this case, I would rename to `rotate_left` and `rotate_right`

**Comment by privat** [comment]
les tests rubix semblent etre trop exigeants pour l'interpreteur et la vm. http://gresil.org/jenkins/view/Int%C3%A9gration/job/nit_tests/engine=niti,label=tolkien/1170/testReport/
est-ce que tu peux les blacklister?

**Comment by privat** [comment]
Ping, so I can merge?

**Comment by privat** [comment]
do not add the extension `.nit` in the skip files. its the testname that is checked.
eg. http://gresil.org/jenkins/job/nit_tests/1174/

**Comment by R4PaSs** [comment]
Reroll: removed `.nit` extension in `.skip` files

**Pull request merged**